pydoctor/driver.py

```diff
@@ -365,6 +365,10 @@ def p(msg):
                 for fn in sorted(system.docstring_syntax_errors):
                     p('    '+fn)
 
+            if system.warnings:
+                # Update exit code if the run has produced warnings.
+                exitcode = 1
```

Comment:
This will break some workflows. Is it in the scope of the PR? 
Anyway it's definitely something to document in the next version changelog!

Reply:
Yes... it will break stuff.

But I think that returning 0 on warnings is a bug :)

I can revert it.... but I wanted to have a simple and useful tox environment that will fail on warnings :)

Reply:
Please use an exit code other than 1, since 1 is returned on uncaught exceptions. I think 3 could be a good choice, so we can reserve 2 for dealing with invalid command line arguments.

Compilers typically return 0 on warnings and only return non-zero when they weren't able to generate code. I think it's useful to have an option (`-Werror` equivalent) to return a non-zero exit code when warnings are reported, but it shouldn't be the default.

When building embedded Linux systems, I have been really annoyed with projects that use `-Werror` in their default release configuration, since it breaks their builds if you use more recent (and therefore often stricter) tooling. We shouldn't inflict that upon anyone if we can avoid it.

Reply:
> Compilers typically return 0 on warnings and only return non-zero when they weren't able to generate code. I think it's useful to have an option (-Werror equivalent) to return a non-zero exit code when warnings are reported, but it shouldn't be the default.

True. I will revert it.

And I think that something like  `-Werror` is better.

After another review it looks like this branch is rather experimental ... with many changes :+1: 
So it might be better to break it into smaller PRs.

I wanted to get the dynamic help ouput...but I was missing the basic dev tools... so I ended up with this PR . Sorry.

Reply:
No problem, it happens to me all the time that when working on one thing, I run into other issues that turn out to be complex enough to become PRs of their own. I still don't have the PR that warns about documented non-existing function parameters in a state where I can submit it...

Reply:
I will move this to https://github.com/twisted/pydoctor/pull/256